### ISSUE TITLE
Ran out of input on slurm cluster multiprocessing

### ISSUE THREAD

**Issue by gulzainali98** [open-mmlab/mmdetection#5169]
Thanks for your error report and we appreciate it a lot.

**Describe the bug**
Whenever I run multi gpu training on my slurm cluster I get following error. 
```
Traceback (most recent call last): File "<string>", line 1, in <module>                                                                                   
File "/opt/conda/lib/python3.6/multiprocessing/spawn.py", line 105, in spawn_main                                         
exitcode = _main(fd)                                                                                                  
File "/opt/conda/lib/python3.6/multiprocessing/spawn.py", line 115, in _main                                             
 self = reduction.pickle.load(from_parent)                                                                           
EOFError: Ran out of input  
/opt/conda/lib/python3.6/multiprocessing/semaphore_tracker.py:143: UserWarning: semaphore_tracker: There appear to be 17 leaked semaphores to clean up at shutdown                                                                                
len(cache))     
```
**Reproduction**
Start distributed training on slurm cluster with python3.6 and using pretrained deformable detr network. 

1. What command or script did you run?

`python3.6 -m torch.distributed.launch --nproc_per_node=4 --master_port=$PORT tools/train.py configs/deformable_detr/deformable_detr_twostage_refine_r50_16x2_50e_coco.py --launcher pytorch ${@:3};`

2. Did you make any modifications to the code or config? Did you understand what you have modified?
 I simply increased samples per GPU to 4 and num_workers to 4 as well. 
3. What dataset did you use?
It's a custom dataset in the coco format
**Environment**

1. Please run `python mmdet/utils/collect_env.py` to collect necessary environment information and paste it here.
TorchVision: 0.8.2+cu110                                                                                                
OpenCV: 3.4.1                                                                                                           
MMCV: 1.3.4                                                                                                             
MMCV Compiler: GCC 7.3                                                                                                  
MMCV CUDA Compiler: 11.0                                                                                                
MMDetection: 2.12.0+931d96e

**Comment by gulzainali98** [comment]
When i train on single GPU it works totally fine. When i train with multiple gpus with num_workers and samples_per_gpu =2 both, i get following error: 

```
/opt/conda/lib/python3.6/multiprocessing/semaphore_tracker.py:143: UserWarning: semaphore_tracker: There appear to be 14 leaked semaphores to clean up at shutdown                                                                                
len(cache)) 
```
There seems to be no way around training mmdetection with multiple gpus.

**Comment by gulzainali98** [comment]
I solved the problem by downgrading pytorch to 1.7.0+cu110 and torchvision to 0.8.0+cu110 It is absolutely necessary to keep mmcv version exactly the same as pytorch version. I install mmcv with following command. 
`pip install mmcv-full -f https://download.openmmlab.com/mmcv/dist/cu110/torch1.7.0/index.html;`
